File: 6KYU/Figurate Numbers #1 - Pentagonal Number/solution.clj
(ns pentnum.core)

(defn p-num [n]
  (-> n
      (* 24)
      inc
      Math/sqrt
      inc
      (/ 6)
      (rem 1)
      zero?))

(defn gp-num [n]
  (zero? (rem (Math/sqrt (+ 1 (* 24 n))) 1)))

(defn sp-num [n]
  (and (p-num n) (zero? (rem (Math/sqrt n) 1))))

File: 6KYU/Figurate Numbers #1 - Pentagonal Number/solution.cs
using System;

public class Penta {
	
	public static Boolean pNum(long n) {
		var r = ((Math.Sqrt(24 * n + 1) + 1) / 6) % 1;
    return r == 0;
	}
	public static Boolean gpNum(long n) {
    return Math.Sqrt((n * 24) + 1) % 1 == 0;
	}
	public static Boolean spNum(long n) {
		return pNum(n) && Math.Sqrt(n) %1 == 0;
	}
}